OpenRA.Mods.Common/Widgets/ScrollPanelWidget.cs

```diff
@@ -347,11 +347,11 @@ public void Bind(IObservableCollection c, Func<object, Widget> makeWidget, Func<
 			{
 				if (collection != null)
 				{
-					collection.OnAdd -= BindingAdd;
-					collection.OnRemove -= BindingRemove;
-					collection.OnRemoveAt -= BindingRemoveAt;
-					collection.OnSet -= BindingSet;
-					collection.OnRefresh -= BindingRefresh;
+					collection.OnAdd -= (item) => BindingAdd(item, collection);
```

Comment:
Derp!  This creates a *new* anonymous delegate and then trys (and failed) to remove it.  The original bindings are therefore never removed.